### PR TITLE
test: rethrow exceptions

### PR DESCRIPTION
**Pull request by vaind**
## :scroll: Description

This adds rehtrows when running in test mode as that seems to have not been finished in previous PRs

Should have no functional changes outside tests and the test changes are only to fix/workaround some issues that were present but hidden before `rethrow` was enabled.

## :bulb: Motivation and Context

Related to https://github.com/getsentry/sentry-dart/issues/1122#issuecomment-1318379866
and #1315

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [x] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: dart/lib/src/sentry_baggage.dart [hunk 1]
```diff
@@ -55,6 +55,7 @@ class SentryBaggage {
           exception: exception,
           stackTrace: stackTrace,
         );
+        // TODO rehtow in automatedTestMode
```
Comment (buenaflor) [inline dart/lib/src/sentry_baggage.dart]: is this comment on purpose or is it an oversight

File: dart/lib/src/sentry_envelope_item.dart [hunk 2]
```diff
@@ -132,6 +132,7 @@ class SentryEnvelopeItem {
       // TODO the data copy could be avoided - this would be most significant with attachments.
       return [...itemHeader, ...newLine, ...data];
     } catch (e) {
+      // TODO rethrow in automatedTestMode
```
Comment (buenaflor) [inline dart/lib/src/sentry_envelope_item.dart]: same comment as the other todo

File: dio/test/mocks.dart [hunk 3]
```diff
@@ -3,6 +3,11 @@ import 'package:sentry/src/transport/rate_limiter.dart';
 
 final fakeDsn = 'https://abc@def.ingest.sentry.io/1234567';
 
+SentryOptions defaultTestOptions() {
+  // ignore: invalid_use_of_internal_member
+  return SentryOptions(dsn: fakeDsn)..automatedTestMode = true;
+}
```
Comment (buenaflor) [inline dio/test/mocks.dart:9]: why do we need another `defaultTestOptions` for dio if we have on in `test_utils`?

Comment (buenaflor) [reply dio/test/mocks.dart:9]: and same question for the other packages, `drift` test and so on

Comment (vaind) [reply dio/test/mocks.dart:9]: it's a different package so yeah

File: dart/lib/src/sentry_envelope_item.dart (hunk 2 above)
Comment (vaind) [reply dart/lib/src/sentry_envelope_item.dart]: Similar to above, except this one actually has another TODO so I believe the refactoring would also solve the try-catch-rethrow sittuation

File: dart/lib/src/sentry_baggage.dart (hunk 1 above)
Comment (vaind) [reply dart/lib/src/sentry_baggage.dart]: I've updated the TODO: we currently don't have options available here and trickling it down would require quite a few changes so left it as a TODO.

File: dio/test/mocks.dart (hunk 3 above)
Comment (buenaflor) [reply dio/test/mocks.dart:9]: got it, what about importing with `import '../../dart/test/test_utils.dart';` ? 

would that work?

Comment (vaind) [reply dio/test/mocks.dart:9]: it does but I'm not a fan of the idea of relative filesystem path imports from other packages, even though it's just for tests. Sounds hacky and not sure it's worth for this (there are worse deduplication candidates)

Comment (buenaflor) [reply dio/test/mocks.dart:9]: sounds good, then let's keep it this way

**Review by buenaflor — APPROVED** [review]

**Pull request merged**